packages/context/src/__tests__/unit/binding-inspector.unit.ts

```diff
@@ -256,3 +257,52 @@ describe('createBindingFromClass()', () => {
     return binding;
   }
 });
+
+describe('isProviderClass', () => {
+  it('tests non-functions', () => {
+    assertNotProviderClasses(undefined, null, 'abc', 1, true, false, {x: 1});
+  });
+
+  it('tests functions that do not have value()', () => {
+    function fn() {}
+    class MyClass {}
+    class MyClassWithVal {
+      value = 'abc';
+    }
+    assertNotProviderClasses(String, Date, fn, MyClass, MyClassWithVal);
+  });
+
+  it('tests functions that have value()', () => {
+    class MyProvider {
+      value() {
+        return 'abc';
+      }
+    }
+
+    class MyAsyncProvider {
+      value() {
+        return Promise.resolve('abc');
+      }
+    }
+
+    function MyJsProvider() {}
+
+    MyJsProvider.prototype.value = function () {
+      return 'abc';
+    };
+
+    assertProviderClasses(MyProvider, MyAsyncProvider, MyJsProvider);
+  });
+
+  function assertNotProviderClasses(...values: unknown[]) {
+    for (const v of values) {
+      expect(isProviderClass(v)).to.be.false();
```

Comment:
So when one of these assertions failed, how are we going to tell which value was considered to be a provider? I think it's better to rewrite this test suite as parameterized tests.

```ts
[undefined, null, 'abc', 1, true, false, {x: 1}].forEach(val => {
  it(`recognizes ${val} as NOT a provider`, () => {
    expect(isProviderClass(val)).to.be.false();
  });
});
```

Same comment applies to other tests too.